dll/win32/shell32/wine/shellpath.c

```diff
@@ -146,90 +146,136 @@ static BOOL WINAPI PathMakeAbsoluteW(LPWSTR path)
 }
 #endif
 
-/* NOTE: GetShortPathName fails if the pathname didn't exist.
-         GetShortPathNameAbsentW should set the short path name that even doesn't exist. */
-static DWORD GetShortPathNameAbsentW(LPCWSTR pszLong, LPWSTR pszShort, DWORD cchShort)
-{
-    FIXME("GetShortPathNameAbsentW(%ls, %p, %ld): stub\n", pszLong, pszShort, cchShort);
-    StringCchCopyW(pszShort, cchShort, pszLong);
-    return lstrlenW(pszShort);
-}
-
-BOOL WINAPI IsLFNDriveW(LPCWSTR lpszPath);
-
-/* @unconfirmed */
+/* @implemented */
 static VOID WINAPI PathQualifyExW(LPWSTR pszPath, LPCWSTR pszDir, DWORD dwFlags)
 {
-    WCHAR szRoot[MAX_PATH], szCopy[MAX_PATH], szCurDir[MAX_PATH];
-    LPWSTR pch;
-    LONG cch;
-    BOOL bCheckLFN;
+    INT iDrive;
+    WCHAR szTemp[MAX_PATH], szRoot[MAX_PATH];
+    LPWSTR pchTemp, pchPath, pchPathEnd;
 
-    if (FAILED(StringCchCopyW(szCopy, _countof(szCopy), pszPath)))
+    /* FIXME: Short pathname */
+
+    /* Copy path to szTemp */
+    if (FAILED(StringCchCopyW(szTemp, _countof(szTemp), pszPath)))
         return;
 
-    FixSlashesAndColonW(szCopy);
+    FixSlashesAndColonW(szTemp); /* every '/' --> '\' */
 
-    if (pszDir)
+    /* Build the root-like path on pszPath, and set pchTemp */
+    if (PathIsUNCW(szTemp)) /* UNC path: Begins with double backslash */
     {
-        cch = GetCurrentDirectoryW(_countof(szCurDir), szCurDir);
-        if (cch <= 0 || cch >= _countof(szCurDir) || !SetCurrentDirectoryW(pszDir))
-            pszDir = NULL;
+        pszPath[2] = UNICODE_NULL; /* Cut off */
```

Comment:
I'm not sure this is always done. There should be an extra condition there...

Reply:
I don't support non-LFN at this moment.

Reply:
Then adding a // FIXME / TODO comment could help...

Reply:
There is that at Line 157.

Reply:
Yes, but it doesn't tell where exactly to fill the code holes.

Reply:
Fixed.